src/main/java/io/vertx/core/http/impl/SharedHttpClient.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.http.impl;
+
+import io.vertx.core.*;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.shareddata.LocalMap;
+
+public class SharedHttpClient extends WrappedHttpClient implements Closeable {
```

Comment:
the client needs to override finalize so it get released when it is gc-ed like the current HTTP client

Reply:
will do

Reply:
Actually HttpClientImpl does not override finalize. Only NetClientImpl does. But it looks ok to me given this is where the resources cleanup needs to happen. Or am I missing something?